app/cdap/components/GridTable/index.tsx

```diff
@@ -24,42 +24,130 @@ import BreadCrumb from 'components/GridTable/components/Breadcrumb';
 import GridHeaderCell from 'components/GridTable/components/GridHeaderCell';
 import GridKPICell from 'components/GridTable/components/GridKPICell';
 import GridTextCell from 'components/GridTable/components/GridTextCell';
+import DirectiveInput from 'components/DirectiveInput';
+import { applyDirectives, getAPIRequestPayload } from 'components/GridTable/services';
 import { useStyles } from 'components/GridTable/styles';
-import {
-  IExecuteAPIResponse,
-  IHeaderNamesList,
-  IParams,
-  IRecords,
-} from 'components/GridTable/types';
 import NoRecordScreen from 'components/NoRecordScreen';
 import LoadingSVG from 'components/shared/LoadingSVG';
 import { IValues } from 'components/WrangleHome/Components/OngoingDataExploration/types';
 import T from 'i18n-react';
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useState, useReducer } from 'react';
 import { useParams } from 'react-router';
 import { flatMap } from 'rxjs/operators';
 import { objectQuery } from 'services/helpers';
+import RecipeSteps from 'components/RecipeSteps';
 import Snackbar from 'components/Snackbar';
+import {
+  IExecuteAPIResponse,
+  IApiPayload,
+  IHeaderNamesList,
+  IGridParams,
+  IObject,
+  IParams,
+  IRecords,
+  IRowData,
+  IMissingListData,
+} from 'components/GridTable/types';
+import styled from 'styled-components';
+import FooterPanel from 'components/FooterPanel';
+import { reducer, initialGridTableState } from 'components/GridTable/reducer';
 import useSnackbar from 'components/Snackbar/useSnackbar';
 
+const TableWrapper = styled(Box)`
+  height: calc(100vh - 193px);
+  overflow-y: auto;
+`;
+
+const TablePanelContainer = styled(Box)`
+  display: flex;
+  font-family: Roboto;
+`;
+
+const RecipeStepPanel = styled(Box)`
+  max-height: calc(100vh - 190px);
+  box-shadow: -3px 4px 15px rgba(68, 132, 245, 0.25);
+`;
+
 export default function GridTable() {
   const { wid } = useParams() as IRecords;
   const params = useParams() as IRecords;
   const classes = useStyles();
+  const { dataprep } = DataPrepStore.getState();
+  enum IGridTableActions {
+    IS_DIRECTIVE_PANEL_OPEN,
+    TABLE_META_INFO,
+  }
+
+  const [gridTableState, dispatch] = useReducer(reducer, initialGridTableState);
+  const { directivePanelIsOpen, tableMetaInfo } = gridTableState;
 
-  const [loading, setLoading] = useState(false);
+  const [loading, setLoading] = useState<boolean>(false);
   const [headersNamesList, setHeadersNamesList] = useState<IHeaderNamesList[]>([]);
-  const [rowsDataList, setRowsDataList] = useState([]);
+  const [rowsDataList, setRowsDataList] = useState<IRowData[]>([]);
```

Comment:
why not move all these state pieces into gridTableState?

Reply:
Done